sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java

```diff
@@ -245,19 +251,25 @@ private void syncList(List<Offer> newList, List<Offer> oldList, OfferType offerT
 					notifyItemRemoved(i, offerType);
 				}
 			}
-		}
 
-		// Add missing offers, the order matters
-		for (int i = 0; i < newList.size(); i++) {
-			Offer offer = newList.get(i);
-			if (i < oldList.size()) {
-				if (!oldList.get(i).equals(offer)) {
-					oldList.add(i, offer);
+			// Add missing offers, the order matters
+			for (int i = 0; i < newList.size(); i++) {
+				Offer offer = newList.get(i);
+				if (i < oldList.size()) {
+					if (!oldList.get(i).equals(offer)) {
+						oldList.add(i, offer);
+						notifyItemInserted(i, offerType);
+					}
+				} else {
+					oldList.add(offer);
 					notifyItemInserted(i, offerType);
 				}
-			} else {
-				oldList.add(offer);
-				notifyItemInserted(i, offerType);
+			}
+		} else {
+			final int size = oldList.size();
+			if (size > 0) {
+				oldList.clear();
```

Comment:
It will be better if syncList will not change its input - it is not clear to the caller of the API that oldList can be cleared

Reply:
It is from day one.
This is the intent for this method, to sync the list.